model/xobject.go

```diff
@@ -528,11 +527,13 @@ func (ximg *XObjectImage) ToImage() (*Image, error) {
 	image.BitsPerComponent = *ximg.BitsPerComponent
 
 	image.ColorComponents = ximg.ColorSpace.GetNumComponents()
+	ximg.primitive.Set("ColorComponents", core.MakeInteger(int64(image.ColorComponents)))
```

Comment:
any particular reason for this?

Reply:
This was a tweak required by the FlateEncoder based on the image. Initialy the `core.FlateEncoder` for images with BPC < 8 didn't add required padding. 
I was looking for any solution that would break the API and wouldn't require additional functions to operate on the results of the encoders. 
This was the solution that actually was the most generic for images.